### PR TITLE
Fix issue where user-provided globs weren't given priority, broke negatives

### LINKED ISSUE #791: exclude some file set in polymer.json works, but `!bower_components/webcomponentsjs/gulpfile.js` not.
### Description
I want exclude some file after `polymer build` and set it in `polymer.json`,
it looks work well, but `!bower_components/webcomponentsjs/gulpfile.js` not work.
 ```javascript
"extraDependencies": [
    "!bower_components/webcomponentsjs/gulpfile.js"
]
```

### Versions & Environment

- Polymer CLI: 1.1.0
- node: 8.0.0
- Operating System: macOS Sierra 10.12.5 (16F73)

#### Steps to Reproduce
1. Create an application project: polymer init polymer-2-application
2. Build: `polymer build`
3. Config: 
```javascript
"extraDependencies": [
    "!bower_components/webcomponentsjs/gulpfile.js"
],
"builds": [
    {
        "bundle": true
    }
]
```
#### Expected Results
`bower_components/webcomponentsjs/gulpfile.js` excluded from dist folder and `build/polymer.json` is correct
```javascript
    "!bower_components/webcomponentsjs/gulpfile.js"
```

#### Actual Results
`bower_components/webcomponentsjs/gulpfile.js` still exists and `build/polymer.json` not correct 
```javascript
    "!/Projects/polymer-demo/bower_components/webcomponentsjs/gulpfile.js"
```
See detail for ndoe v8.0.0 https://travis-ci.org/Polymer/polymer-cli/jobs/240647279#L832
and for node v6.11.0 detail https://travis-ci.org/Polymer/polymer-cli/jobs/240647280#L832

### PR DESCRIPTION
**Pull request by FredKSchott**
Fixes #791

### CODE REVIEW

**Comment by usergenic** [comment]
@FredKSchott any idea what's up with the appveyor failure for the branch?  https://ci.appveyor.com/project/justinfagnani/polymer-cli/build/1.0.1539

Doesn't seem like it could be related, but maybe there was something introduced in master...?

File: src/polymer-cli.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export class PolymerCli {
     this.defaultConfigOptions = this.defaultConfigOptions || {};
     this.defaultConfigOptions.extraDependencies =
         this.defaultConfigOptions.extraDependencies || [];
-    this.defaultConfigOptions.extraDependencies.push(
+    this.defaultConfigOptions.extraDependencies.unshift(
```
Comment (baocang) [inline src/polymer-cli.ts]: Is negatives expression needs hight priority by default likes below?
```javascript
this.defaultConfigOptions.extraDependencies.sort(function(first, second) {
    console.log(first, second);
    if (first && '!' === first[0]) { return 1; }
    if (second &&'!' === second[0]) { return 0; }
    return 0;
});
‘‘‘

**Comment by FredKSchott** [comment]
screen shot 2017-07-13 at 12 36 49 pm

I thought I had gotten this merged...

File: src/polymer-cli.ts (hunk 1 above)
Comment (FredKSchott) [inline src/polymer-cli.ts]: no, we still want to preserve order of the extraDependencies you give us

**Pull request merged**